homeassistant/components/ping/device_tracker.py

```diff
@@ -42,123 +36,95 @@
-            for (dev_id, ip) in config[CONF_HOSTS].items()
-        ]
-
-        async def async_update(now: datetime) -> None:
-            """Update all the hosts on every interval time."""
-            results = await gather_with_limited_concurrency(
-                CONCURRENT_PING_LIMIT,
-                *(hass.async_add_executor_job(host.update) for host in hosts),
-            )
-            await asyncio.gather(
-                *(
-                    async_see(dev_id=host.dev_id, source_type=SourceType.ROUTER)
-                    for idx, host in enumerate(hosts)
-                    if results[idx]
-                )
-            )
 
-    else:
+class PingDeviceTracker(ScannerEntity):
+    """Representation of a Ping device tracker."""
 
-        async def async_update(now: datetime) -> None:
-            """Update all the hosts on every interval time."""
-            responses = await async_multiping(
-                list(ip_to_dev_id),
-                count=PING_ATTEMPTS_COUNT,
-                timeout=ICMP_TIMEOUT,
-                privileged=privileged,
-            )
-            _LOGGER.debug("Multiping responses: %s", responses)
-            await asyncio.gather(
-                *(
-                    async_see(dev_id=dev_id, source_type=SourceType.ROUTER)
-                    for idx, dev_id in enumerate(ip_to_dev_id.values())
-                    if responses[idx].is_alive
-                )
-            )
+    config_entry: ConfigEntry
+    ping: PingDataSubProcess | PingDataICMPLib
 
-    async def _async_update_interval(now: datetime) -> None:
-        try:
-            await async_update(now)
-        finally:
-            if not hass.is_stopping:
-                async_track_point_in_utc_time(
-                    hass, _async_update_interval, now + interval
-                )
+    def __init__(
+        self,
+        name: str,
+        config_entry: ConfigEntry,
+        ping_cls: PingDataSubProcess | PingDataICMPLib,
+    ) -> None:
+        """Initialize the Ping device tracker."""
+        super().__init__()
+
+        self._attr_name = name
+        self._unique_id = f"{config_entry.entry_id}_device_tracker"
+        self.config_entry = config_entry
+        self.ping = ping_cls
+
+    @property
+    def unique_id(self) -> str:
+        """Return unique ID of the entity."""
+        return self._unique_id
+
+    @property
+    def source_type(self) -> SourceType:
+        """Return the source type which is router."""
+        return SourceType.ROUTER
+
+    @property
+    def is_connected(self) -> bool:
+        """Return true if ping returns is_alive."""
+        return self.ping.is_alive
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if entity is enabled by default."""
+        if CONF_IMPORTED_BY in self.config_entry.options:
+            return bool(self.config_entry.options[CONF_IMPORTED_BY] == "device_tracker")
+        return False
```

Comment:
I'm pretty sure this platform supports shorthand `_attr` as well

Reply:
Yes, `_attr_unique_id` and `_attr_ entity_registry_enabled_default` are supported.

Reply:
I need to use the property attributes (https://developers.home-assistant.io/docs/core/entity/device-tracker), the shorthand `_attr` are not supported.